ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/home/model/WalletSnapshot.kt

```diff
@@ -13,14 +14,33 @@ data class WalletSnapshot(
     val orchardBalance: WalletBalance,
     val saplingBalance: WalletBalance,
     val transparentBalance: WalletBalance,
-    val pendingCount: Int
+    val pendingCount: Int,
+    val progress: Int,
+    val synchronizerError: SynchronizerError?
 ) {
     // Note: the wallet is effectively empty if it cannot cover the miner's fee
     val hasFunds = saplingBalance.available.value >
         (ZcashSdk.MINERS_FEE.value.toDouble() / Zatoshi.ZATOSHI_PER_ZEC) // 0.00001
     val hasSaplingBalance = saplingBalance.total.value > 0
 
     val isSendEnabled: Boolean get() = status == Synchronizer.Status.SYNCED && hasFunds
+
+    val hasSynchronizerError: Boolean = synchronizerError != null
```

Comment:
I'm not sure if we need this boolean property—I have a comment further below showing where this property might be making our code more complex and a bit less safe.